transport/src/main/java/io/netty/channel/DefaultChannelPipeline.java

```diff
@@ -582,34 +745,18 @@ private void callHandlerAdded0(final AbstractChannelHandlerContext ctx) {
         }
     }
 
-    private void callHandlerRemoved(final AbstractChannelHandlerContext ctx) {
-        if (isExecuteLater(ctx)) {
-            ctx.executor().execute(new OneTimeTask() {
-                @Override
-                public void run() {
-                    callHandlerRemoved0(ctx);
-                 }
-            });
-            return;
-        }
-        callHandlerRemoved0(ctx);
-    }
-
     private void callHandlerRemoved0(final AbstractChannelHandlerContext ctx) {
         // Notify the complete removal.
         try {
             ctx.handler().handlerRemoved(ctx);
             ctx.setRemoved();
         } catch (Throwable t) {
+            ctx.setRemoved();
```

Comment:
needs to happen before we invoke fireExceptionCaught or could we move it to a finally block to avoid the double call?